### DISCUSSION TITLE
Cnb tomcat container

### DISCUSSION THREAD

**Discussion by manojchander6** [paketo-buildpacks/feedback#33] [Q&A]
I  have 3  cnb pack based  tomcats  containers 
Hosting my applications.

I am looking to containerize it into single container with cnb build packs . 

Is it possible to run multiple tomcats inside single container  .
Basically have 3 workspaces (webapps) folders and deploy our wars files .

**Comment by dmikusa** [comment]
It's not possible to run multiple instances of Tomcat in a single container with Paketo Buildpacks, nor would I recommend doing that. You should stick to one process per container, that is a pretty widely recommended practice (with the exception of side-cars). 

There is a [feature request to run multiple WAR applications in a single Tomcat instance](https://github.com/paketo-buildpacks/apache-tomcat/issues/134). When that issue is implemented, you could deploy multiple applications within a single Tomcat process each on a separate context path all in the same container. The main difference is that there's only one Tomcat process, so while I don't think it's optimal it's not violating the rule above.

Feel free to subscribe to that issue to follow progress.
